### PR TITLE
Update  README.md and CONTRIBUTING.md with new content sections and logo. Upgrade pre-commit hooks and build-tools.

### PR DESCRIPTION
**Pull request by soerenmartius**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,9 +101,11 @@ module "bucket" {
...
 ## Module Argument Reference
+
 See [variables.tf](variables.tf) and [examples/](examples) for details and use-cases.
 
-#### Module Configuration
+### Module Configuration
+
 - **`module_enabled`**: *(Optional `bool`)*
 Specifies whether resources in the module will be created.
 Default is `true`.
```
Comment (mariux) [inline README.md:111]:
```suggestion
- **`module_enabled`**: *(Optional `bool`)*

Specifies whether resources in the module will be created.
Default is `true`.
```

File: README.md [hunk 2]
```diff
@@ -424,40 +465,53 @@ containing all arguments as specified above and the other attributes as specifie
...
+
+### Backwards compatibility in `0.0.z` and `0.y.z` version
 
-#### Backwards compatibility in `0.0.z` and `0.y.z` version
-- Backwards compatibility in versions `0.0.z` is **not guaranteed** when `z` is increased. (Initial development)
-- Backwards compatibility in versions `0.y.z` is **not guaranteed** when `y` is increased. (Pre-release)
+- In the context of initial development, backwards compatibility in versions `0.0.z` is **not guaranteed** when `z` is
+  increased. (Initial development)
+- In the context of pre-release, backwards compatibility in versions `0.y.z` is **not guaranteed** when `y` is
+  increased. (Pre-release)
```
Comment (mariux) [inline README.md:482]: the old wording was actually way more readable as this is really important to make this clear to the point... now the context of the versions changed is harder to get as it is not visually comparable anymore... @soerenmartius we should definitely revert this back..

File: README.md [hunk 3]
```diff
@@ -424,40 +465,53 @@ containing all arguments as specified above and the other attributes as specifie
...
-1) `MAJOR` version when we make incompatible changes,
-2) `MINOR` version when we add functionality in a backwards compatible manner, and
-3) `PATCH` version when we make backwards compatible bug fixes.
+Using the given version number of `MAJOR.MINOR.PATCH`, we apply the following constructs:
+
+1. Use the `MAJOR` version for incompatible changes.
+1. Use the `MINOR` version when adding functionality in a backwards compatible manner.
+1. Use the `PATCH` version when introducing backwards compatible bug fixes.
+
+### Backwards compatibility in `0.0.z` and `0.y.z` version
```
Comment (mariux) [inline README.md:477]: this wording actually also makes no sense...  what does "USE" mean... this was not caught in the review as too many changes were applied i guess... the original wording is 1:1 form the semantic versioning documentation... so why change it at all?

File: README.md (hunk 2 above)
Comment (mariux) [reply README.md:482]: opened a pr for this

File: README.md (hunk 3 above)
Comment (mariux) [reply README.md:477]: opened a pr for this

**Comment by soerenmartius** [comment]
r4r @mariux

**Pull request merged**